### PR TITLE
Enable Solana feature flag by default on Desktop

### PR DESCRIPTION
**Pull request by yrliou**
Resolves https://github.com/brave/brave-browser/issues/22348

Security review: https://github.com/brave/security/issues/827

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/about_flags.cc
Comment (yrliou) [inline browser/about_flags.cc:380]: Use kOsAndroid directly to avoid confusion.
See https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/about_flags.cc;l=295 in upstream

**Comment by yrliou** [comment]
convert to draft for now as I need to fix some tests

**Review by darkdh — APPROVED** [review]
wallet changes lgtm, I will defer the final decision to security team

File: components/brave_wallet/resources/solana_provider.js
Comment (kdenhartog) [inline components/brave_wallet/resources/solana_provider.js:32]: Confirming this version hasn't been updated and is still running v1.36.0 which I reviewed in PR #12554

**Review by kdenhartog — DISMISSED** [review]
Code all looks good to me. From my perspective this is good to merge

**Comment by kdenhartog** [comment]
One thing popped in my head after I finished the review. Do we need to wrap this code in an `is_ios` build flag as well so that  it's not on by default there as well? I believe they're still working through some features based on ongoing discussions in https://github.com/brave/security/issues/811

**Comment by yrliou** [comment]
AFAIK, this feature flag is currently used by Android and Desktop only.
I'm not sure what's iOS's plan on developing Solana support but I believe it's not started yet.

> One thing popped in my head after I finished the review. Do we need to wrap this code in an `is_ios` build flag as well so that it's not on by default for the iOS platform as well? I believe they're still ~working through some features based on ongoing discussions in [brave/security#811](https://github.com/brave/security/issues/811)~ needing to implement Solana support and it's unclear if this would enable it on iOS by default for them.

**Comment by yrliou** [comment]
I believe no actions in this PR should be required for iOS for now, but cc @kylehickinson just in case.

**Comment by kylehickinson** [comment]
@yrliou yep, no action required for iOS. Looks like it's enabled by default for us which is what we do for feature flags since we don't support

**Review by kdenhartog — APPROVED** [review]
Thanks for confirmation @kylehickinson and @yrliou re-approving

**Comment by yrliou** [comment]
Failed unit tests are irrelevant to this PR, already fixed in master by https://github.com/brave/brave-core/pull/13067, merging.

**Pull request merged**
